src/atomate2/common/jobs/anharmonicity.py

```diff
@@ -0,0 +1,611 @@
+"""Jobs for running anharmonicity quantification."""
+
+from __future__ import annotations
+
+import contextlib
+import logging
+from typing import TYPE_CHECKING
+
+import numpy as np
+from jobflow import Flow, Response, job
+from phonopy import Phonopy
+from pymatgen.core import Structure
+from pymatgen.core.units import kb
+from pymatgen.io.phonopy import get_phonopy_structure, get_pmg_structure
+
+from atomate2.aims.schemas.calculation import Calculation
+from atomate2.common.schemas.anharmonicity import AnharmonicityDoc
+
+if TYPE_CHECKING:
+    from pathlib import Path
+    from typing import Any
+
+    from atomate2.aims.jobs.base import BaseAimsMaker
+    from atomate2.common.schemas.phonons import ForceConstants, PhononBSDOSDoc
+    from atomate2.forcefields.jobs import ForceFieldStaticMaker
+    from atomate2.vasp.jobs.base import BaseVaspMaker
+
+
+logger = logging.getLogger(__name__)
+
+# Constants
+EV = 1.60217733e-19  # [J]
+AA = 1e-10  # [m]
+AMU = 1.66053904e-27  # [kg]
+THZ = 1e12  # [1/s]
+PI = np.pi
+OMEGA_TO_THZ = (EV / AA**2 / AMU) ** 0.5 / THZ / 2 / PI  # 15.633302 THz
+
+
+@job
+def get_phonon_supercell(phonon_doc: PhononBSDOSDoc) -> Structure:
+    """Get the phonon supercell of a structure.
+
+    Parameters
+    ----------
+    phonon_doc: PhononBSDOSDoc
+        The output of a phonopy workflow
+
+    Returns
+    -------
+    Structure
+        The phonopy structure
+    """
+    cell = get_phonopy_structure(phonon_doc.structure)
+    phonon = Phonopy(
+        cell,
+        phonon_doc.supercell_matrix,
+    )
+    return get_pmg_structure(phonon.supercell)
+
+
+def get_sigma_per_atom(
+    structure: Structure,
+    forces_dft: np.ndarray,
+    forces_harmonic: np.ndarray,
+) -> list[tuple[str, float]]:
+    """Compute the atom-resolved sigma^A measure.
+
+    Parameters
+    ----------
+    structure: Structure
+        The structure to use for the calculation
+    forces_dft: np.ndarray
+        DFT calculated forces
+    forces_harmonic: np.ndarray
+        Harmonic approximation of the forces
+
+    Returns
+    -------
+    list[tuple[str, float]]
+        List of tuples in the form (atom symbol, sigma^A)
+        for all the atoms in the structure.
+    """
+    # Ensure that DFT and harmonic forces are in np format
+    forces_dft = np.array(forces_dft)
+    forces_harmonic = np.array(forces_harmonic)
+
+    # Check shapes of forces
+    if len(np.shape(forces_dft)) == 2:
+        forces_dft = np.expand_dims(forces_dft, axis=0)
+        forces_harmonic = np.expand_dims(forces_harmonic, axis=0)
+
+    atom_numbers = np.array([site.specie.number for site in structure.sites])
+    if np.shape(forces_dft)[1] == 3 * structure.num_sites:
+        atom_numbers = atom_numbers.repeat(3)
+
+    symbols = np.array([site.specie.name for site in structure.sites])
+    unique_atoms = np.unique(atom_numbers)
```

Comment:
It's more about elemental anharmonicities or am I wrong? Might be good to emphasize this. I would have expected an anharmonicty score for each unique site in the structure.

Reply:
You are correct these are elemental values, in principle we could also add site values as well.

Reply:
I would recommend making this clearer in the documentation/naming of the methods.

I could imagine that such site-wise values should be very insightful for typical structures with rattling atoms.